### PR TITLE
Quadruna Heartbeat Monitor, and Car Specific Shared Directories

### PR DESCRIPTION
**Pull request by liam-ilan**
### Summary
Set up a separate heartbeat monitor for Thruna and Quadruna.

### Changelist 
- Created car specific shared directories in `firmware/thruna` and `firmware/quadruna`
- Moved heartbeat board enums to respective `shared` dirs

### Testing Done
- [x] Flashed on Thruna.

### Resolved Issues
None

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

**PR > 500 lines, but should be merged in one go, don't want things to get fragmented (it's all one feature).**

### CODE REVIEW

File: firmware/thruna/shared/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
+if("${TARGET}" STREQUAL "test")
+    list(APPEND HDRS_TO_FAKE 
+        "${SHARED_IO_INCLUDE_DIR}/io_time.h"
+        "${SHARED_IO_INCLUDE_DIR}/io_buzzer.h"
+        "${SHARED_IO_INCLUDE_DIR}/io_led.h"
+        "${SHARED_IO_INCLUDE_DIR}/io_rgbLed.h"
+        "${SHARED_IO_INCLUDE_DIR}/io_switch.h"
+    )
```
Comment (gtaharaedmonds) [inline firmware/thruna/shared/CMakeLists.txt:8]: Now that all the test source files are shared, do you think we need this CMake code?

File: .vscode/launch.json [hunk 2]
```diff
@@ -454,12 +454,40 @@
...
+    {
+      "name": "Debug Tests: Shared (Thruna)",
+      "type": "cppdbg",
+      "request": "launch",
+      "program": "${workspaceRoot}/build_fw_test/firmware/thruna/shared/thruna_shared_test",
+      "cwd": "${workspaceRoot}",
+      "preLaunchTask": "Build Tests: Shared (Thruna)",
       "linux": {
         "MIMode": "gdb"
       },
```
Comment (gtaharaedmonds) [inline .vscode/launch.json:493]: Do we need seperate shared test run configs for Thruna/Quadruna now? (really same comment as below)

**Review by gtaharaedmonds — CHANGES_REQUESTED** [review]

File: firmware/quadruna/BMS/CMakeLists.txt [hunk 3]
```diff
@@ -6,22 +6,22 @@ set(LINKER_SCRIPT "${LINKER_DIR}/stm32h733vgtx/stm32h733vgtx_app.ld")
 file(GLOB_RECURSE APP_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/src/app/*.c")
 list(APPEND APP_SRCS 
     "${SHARED_APP_INCLUDE_DIR}/app_stateMachine.c"
-    "${SHARED_APP_INCLUDE_DIR}/app_heartbeatMonitor.c"
     "${SHARED_APP_INCLUDE_DIR}/app_timer.c"
     "${SHARED_APP_INCLUDE_DIR}/app_math.c"
+    "${SHARED_APP_INCLUDE_DIR}/app_heartbeatMonitor.c"
```
Comment (Lucien950) [inline firmware/quadruna/BMS/CMakeLists.txt:11]: cheeky

File: firmware/thruna/shared/CMakeLists.txt (hunk 1 above)
Comment (Lucien950) [reply firmware/thruna/shared/CMakeLists.txt:8]: and if we need the cmake code, can we move it under shared/tests/CMakeLists.txt and include it conditionally, cause this is confusing

Comment (liam-ilan) [reply firmware/thruna/shared/CMakeLists.txt]: Removed!

File: .vscode/launch.json (hunk 2 above)
Comment (liam-ilan) [reply .vscode/launch.json]: Also removed

**Review by gtaharaedmonds — APPROVED** [review]
LGTM

**Pull request merged**